rholang/src/test/scala/coop/rchain/rholang/interpreter/matcher/MatchTest.scala

```diff
@@ -102,6 +106,17 @@ class VarMatcherSpec extends FlatSpec with Matchers {
     assertSpatialMatch(target, pattern, Some(Map[Int, Par](0 -> GInt(7), 1 -> GInt(8))))
   }
 
+  "Matching huge list of targets with no patterns and a reminder" should "better be quick" in {
+    //This is a very common case in rspace that can be handled in linear time, yet was quadratic for a short while
+    val target: Par = Par(exprs = Seq.fill(1000)(GInt(1): Expr))
+    val pattern: Par = EVar(FreeVar(0))
+
+    import org.scalatest.time.SpanSugar._
+    failAfter(5 seconds) {
```

Comment:
Doest this take ~3s on your laptop?

Reply:
less than 1 s, and more than 15 s with the quadratic regression

Reply:
OK. I fear this might fail in CI from time to time

Reply:
So do I, but we can then bump the timeout to 10s or so - that should still differentiate a quadratic regression reliably